src/easylink/step.py

```diff
@@ -1650,9 +1691,45 @@ def _configure_subgraph_steps(self) -> None:
         This method recursively traverses the ``StepGraph`` and sets the configuration
         state for each ``Step`` until reaching all leaf nodes.
         """
-        nodes = self._step.step_graph.nodes
-        for node in nodes:
-            step = nodes[node]["step"]
+        for node in self._nodes:
+            step = self._nodes[node]["step"]
+            # NOTE: The subgraph may or may not have the step name as an outer key at
+            # this point (e.g. I/O nodes, the chosen step from a ChoiceStep, etc.)
+            step_config = self.step_config.get(step.name, self.step_config)
```

Comment:
I don't hate this, but would like to expand this comment to explain these examples more. I/O nodes have no configuration, so I'm not sure how the configuration state works there. Currently ChoiceSteps choose between subgraphs, not between steps, so I'm not sure what the "step name" would mean there.

Reply:
Nice, the new version looks much easier to understand!

Reply:
I can't lie, it isn't obvious to me why IOSteps even _have_ a configuration state 🤔

Reply:
I'm reasonably sure just for backward compatibility with other `Steps`.

Reply:
But maybe @patricktnast can clarify if he remembers

Reply:
Perhaps it would be clearer to assign them a `NullConfigurationState`, sort of like how we assign them `NullImplementation`s. Here we pass in (IIUC) the entire configuration under `steps` and I presume that is never used.

Reply:
I looked briefly at this and I think just like how in [MIC-5897](https://github.com/ihmeuw/easylink/pull/158) we are discussing how slot mappings are used for both subgraphs as well as mapping from step to implementation nodes, we need much (but perhaps not all) of the ConfigurationState methods to exist even for IOSteps (e.g. we still need go `LeafConfigurationState.get_implementation_edges()` which does rely on the step config.